### ISSUE TITLE
Could you kindly consider unifying the writing styles for the usage of settings?

### ISSUE THREAD

**Issue by fiyen** [Future-House/paper-qa#492]
This is an excellent project, but the manual setting of APIs seems overly complicated. The same settings do not produce the same effect across different commands, which makes it very difficult for beginners. For example, the `llm` settings in `doc.query` can be written as:

```python
llm_config = dict(
    model_list=[
        dict(
            model_name='gpt-4o-mini',
            litellm_params=dict(
                model='gpt-4o-mini',
                api_key="xxx",
                api_base="https://"
            )
        ),
    ],
)
```

However, the `embedding` settings must be written like this to be effective:

```python
embedding_config = dict(
    model_list=[
        dict(
            model_name='text-embedding-3-small',
            litellm_params=dict(
                model='text-embedding-3-small',
                api_key="xxx",
                api_base="xxx"
            )
        )
    ],
    api_key="xxx",
    api_base="https://xxxx"
)
```

This is because when the `embedding` requests `api_key` and `api_base`, it does not use the parameters inside `model_list`. Even more frustrating is that when using `doc.query`, the settings can be written like this:

```python
settings = Settings(
    llm='gpt-4o-mini',
    summary_llm='gpt-4o-mini',
    llm_config=llm_config,
    summary_llm_config=llm_config,
    embedding='text-embedding-3-small',
    embedding_config=embedding_config
)
```

However, the same writing style does not work in `ask` because the way `api_key` and `api_base` are indexed is different. Admittedly, this is due to the different usage modes of the tools, but having different settings in the same project can cause a lot of confusion. It would be great to unify the writing style of the settings.

**Comment by fiyen** [comment]
If the value of api related params are not given in some modules, these values may be set to the global_xx values.

**Comment by jamesbraza** [comment]
Yeah @fiyen we have a lot of work to do here. We pushed hard to open source a huge amount in a very small window of time, and are going back and improving everything over time
